game/players.py

```diff
@@ -33,66 +40,73 @@ def __init__(self, Q = False, auto=False):
             self.CURR_STATE = 0
 
             self.PREV_REWARD = 0
+
+            #constant reqards given to the agent after completion/fold
             self.COMPL_REWARD = 10.0
             #self.PLAY_REWARD defined later
             self.DRAW_PENALTY = (-0.5)
             #self.FOLD_PENALTY defined later
 
     def init(self):
-        self.hand = []
-        self.activate()
+        self.hand = [] #This is for emptying everything?
+        self.activate() #what is tis for?
+
 
+    #called when player folds
     def deactivate(self):
         self.active = False
 
+    #is this called anywhere?
     def activate(self):
         self.active = True
 
+    #where is this function called?
     def bot(self, Q):
         self.isbot = True
-        self.isQbot = Q
+        self.isQbot = Q 
 
+    #drawing a card from the shuffled main deck pile
     def draw(self, deck):
         self.hand.append(deck.main_pile.pop())
 
+    #calculate the score of the cards in hand
     def calc_score(self):
+
         if not len(self.hand):
             if not self.score:
                 self.score = 0
             # TODO: Ideally the following should be a choice
             elif self.score < 10:
-                self.score = self.score - 1
+                self.score = self.score - 1 #is this correct?
             else:
                 self.score = self.score - 10
+        
         else:
-            uniq_hand = list(set(self.hand))
-            increment = sum(map(lambda x: x if x < 7 else 10, uniq_hand))
+            uniq_hand = list(set(self.hand)) #arrange the cards in alphabetical order and removes multiple cards for a number
+            increment = sum(map(lambda x: x if x < 7 else 10, uniq_hand)) # value same as card except for x=7 i.e. lama card
             self.score = self.score + increment
         return self.score
 
     def delete(self, n):
-        i = 0
-        while i < len(self.hand):
-            if self.hand[i] == n:
-                return self.hand.pop(i)
-            i = i + 1
+        if n in self.hand:
+            return self.hand.pop(self.hand.index(n))
+        else:
+            print("card not in hand")
 
+    #where is this called??
```

Comment:
In `core.py`, its called when the bot is being tested and when the bot is being added into a real game using the button (line 487 in `core.py`).